contrib/python-bindings/source/tria_accessor_wrapper.cc

```diff
@@ -0,0 +1,317 @@
+    }
+  }
+
+
+
+  TriaAccessorWrapper::TriaAccessorWrapper(const TriaAccessorWrapper &other)
+    :
+    structdim(other.structdim),
+    dim(other.dim),
+    spacedim(other.spacedim)
+  {
+    if ((dim == 2) && (spacedim == 2) && (structdim == 1))
+      {
+        TriaAccessor<1,2,2> *other_accessor =
+          static_cast<TriaAccessor<1,2,2>*>(other.tria_accessor);
+        tria_accessor = new TriaAccessor<1,2,2>(*other_accessor);
+      }
+    else if ((dim == 2) && (spacedim == 3) && (structdim == 1))
+      {
+        TriaAccessor<1,2,3> *other_accessor =
+          static_cast<TriaAccessor<1,2,3>*>(other.tria_accessor);
+        tria_accessor = new TriaAccessor<1,2,3>(*other_accessor);
+      }
+    else if ((dim == 3) && (spacedim == 3) && (structdim == 2))
+      {
+        TriaAccessor<2,3,3> *other_accessor =
+          static_cast<TriaAccessor<2,3,3>*>(other.tria_accessor);
+        tria_accessor = new TriaAccessor<2,3,3>(*other_accessor);
+      }
+    else
+      AssertThrow(false, ExcMessage("Wrong structdim-dim-spacedim combination."));
+  }
+
+  TriaAccessorWrapper::TriaAccessorWrapper(void* tria_accessor, 
+                                           const int structdim, 
+                                           const int dim, 
+                                           const int spacedim):
+    structdim(structdim),
+    dim(dim),
+    spacedim(spacedim),
+    tria_accessor(tria_accessor)
+  {}
+
+
+  TriaAccessorWrapper::~TriaAccessorWrapper()
+  {
+    if (dim != -1)
+      {
+        if ((dim == 2) && (spacedim == 2) && (structdim == 1))
+          {
+            // We cannot call delete on a void pointer so cast the void pointer back
+            // first.
+            TriaAccessor<1,2,2> *tmp =
+              static_cast<TriaAccessor<1,2,2>*>(tria_accessor);
+            delete tmp;
+          }
+        else if ((dim == 2) && (spacedim == 3) && (structdim == 1))
+          {
+            TriaAccessor<1,2,3> *tmp =
+              static_cast<TriaAccessor<1,2,3>*>(tria_accessor);
+            delete tmp;
+          }
+        else
+          {
+            TriaAccessor<2,3,3> *tmp =
+              static_cast<TriaAccessor<2,3,3>*>(tria_accessor);
+            delete tmp;
+          }
+
+        dim = -1;
+        spacedim = -1;
+	structdim = -1;
+        tria_accessor = nullptr;
+      }
+  }
+
+  PointWrapper TriaAccessorWrapper::get_barycenter() const
+  {
+    if ((dim == 2) && (spacedim == 2) && (structdim == 1))
+      return internal::get_barycenter<1,2,2>(tria_accessor);
+    else if ((dim == 2) && (spacedim == 3) && (structdim == 1))
+      return internal::get_barycenter<1,2,3>(tria_accessor);
+    else
+      return internal::get_barycenter<2,3,3>(tria_accessor);
+  }
+
+
+
+  void TriaAccessorWrapper::set_boundary_id(const int boundary_id)
+  {
+    AssertThrow(static_cast<types::material_id>(boundary_id) < numbers::invalid_boundary_id,
+                ExcMessage("boundary_id is too large."));
```

Comment:
I don't think this `AssertThrow` make sense. I think that `numbers::invalid_bounda_id` is `unsigned int (-1)` which is larger than any integer.